src/analysis.ts

```diff
@@ -12,16 +12,309 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {DocumentDescriptor} from './ast/ast';
-import {AnalyzedPackage} from './serialized-analysis';
+import * as fs from 'fs';
+import * as jsonschema from 'jsonschema';
+import * as path from 'path';
+
+import {Descriptor, DocumentDescriptor, ElementDescriptor, InlineDocumentDescriptor, PropertyDescriptor} from './ast/ast';
+import {JsonDocument} from './json/json-document';
+import {Document} from './parser/document';
+import {AnalyzedPackage, Attribute, Element, Event, Property} from './serialized-analysis';
+
+const validator = new jsonschema.Validator();
+const schema = JSON.parse(
+    fs.readFileSync(path.join(__dirname, 'analysis.schema.json'), 'utf-8'));
+
+export class ValidationError extends Error {
+  errors: jsonschema.ValidationError[];
+  constructor(result: jsonschema.ValidationResult) {
+    const message = `Unable to validate serialized Polymer analysis. ` +
+        `Got ${result.errors.length} errors: ` +
+        `${result.errors.map(err => '    ' + (err.message || err)).join('\n')}`;
+    super(message);
+    this.errors = result.errors;
+  }
+}
+
 
 export class Analysis {
   private _descriptors: DocumentDescriptor[];
+
   constructor(descriptors: DocumentDescriptor[]) {
     this._descriptors = descriptors;
   }
 
-  serialize(): AnalyzedPackage {
-    throw new Error('not implemented');
+  /**
+   * Throws if the given object isn't a valid AnalyzedPackage according to
+   * the JSON schema.
+   */
+  static validate(analyzedPackage: AnalyzedPackage) {
+    const result = validator.validate(analyzedPackage, schema);
+    if (result.throwError) {
+      throw result.throwError;
+    }
+    if (result.errors.length > 0) {
+      throw new ValidationError(result);
+    }
+    if (!/^1\.\d+\.\d+$/.test(analyzedPackage.schema_version)) {
+      throw new Error(
+          `Invalid schema_version in AnalyzedPackage. ` +
+          `Expected 1.x.x, got ${analyzedPackage.schema_version}`);
+    }
+  }
+
+  serialize(packagePath?: string): AnalyzedPackage {
```

Comment:
I still think this should be separate from Analysis, aa function that takes an Analysis